src/parcels/_core/fieldset.py

```diff
@@ -257,6 +259,47 @@ def from_copernicusmarine(ds: xr.Dataset):
 
         return FieldSet(list(fields.values()))
 
+    def from_fesom2(ds: ux.UxDataset):
+        """Create a FieldSet from a FESOM2 uxarray.UxDataset.
+
+        Parameters
+        ----------
+        ds : uxarray.UxDataset
+            uxarray.UxDataset as obtained from the uxarray package.
+
+        Returns
+        -------
+        FieldSet
+            FieldSet object containing the fields from the dataset that can be used for a Parcels simulation.
+        """
+        ds = ds.copy()
+        ds_dims = list(ds.dims)
+        if not all(dim in ds_dims for dim in ["time", "nz", "nz1"]):
+            raise ValueError(
+                f"Dataset missing one of the required dimensions 'time', 'nz', or 'nz1'. Found dimensions {ds_dims}"
+            )
+        grid = UxGrid(ds.uxgrid, z=ds.coords["nz"])
+        ds = _discover_fesom2_U_and_V(ds)
+
+        fields = {}
+        if "U" in ds.data_vars and "V" in ds.data_vars:
+            # TO DO: Assign interpolator based on dimensions
```

Comment:
Indeed, I would set the Interpolator here (rather than in Field)

Reply:
Removed this comment in https://github.com/Parcels-code/Parcels/pull/2319/commits/1261d2390c054b03a7f61f2cd3cc5b0c19ff033e